extensions/ingestion/IngestionQueuePopulator.js

```diff
@@ -0,0 +1,85 @@
+const async = require('async');
+const QueuePopulatorExtension =
+          require('../../lib/queuePopulator/QueuePopulatorExtension');
+
+const zookeeper = require('../../lib/clients/zookeeper');
+class IngestionQueuePopulator extends QueuePopulatorExtension {
+    constructor(params) {
+        super(params);
+        this.config = params.config;
+    }
+
+    createZkPath(cb, source) {
+        const { zookeeperPath } = this.extConfig;
+        const pathArray = [];
+        this.bucketPrefix = source.prefix ? source.prefix : source.name;
+        if (source.raftCount) {
+            for (let i = 1; i <= source.raftCount; i++) {
+                const path = `/ingestion/${source.name}/provisions/${i}`;
+                pathArray.push(path);
```

Comment:
To avoid some complexity, you could just call `this.zkClient.mkdirp()` in an `async.times()` loop: no need to call `getData()` first if you ignore `NODE_EXISTS` errors from the result of `mkdirp`, and `async.times` callback is given the current index in the loop that you can use to make the path that you pass to `mkdirp`.